hw/top_earlgrey/data/ip/chip_csrng_testplan.hjson

```diff
@@ -0,0 +1,101 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  name: chip_csrng
+  testpoints: [
+    // CSRNG (pre-verified IP) integration tests:
+    {
+      name: chip_sw_csrng_edn_cmd
+      desc: '''Verify incoming command interface from EDN.
+
+            - Have each EDN instance issue an instantiate, reseed and generate command to CSRNG.
+            - On each command done, verify the reception of edn cmd req done interrupt.
+            - Run OTBN randomness test to test the output from EDN0 and EDN1.
+            - Check the data returned to EDN via connectivity assertion checks.
+
+            Notes for silicon targets:
+            - The test bench should program the Ibex to perform the tasks above.
+            - There also needs to be a way for Ibex to communicate the test pass/fail to the test infrastructure via an external interface, such as UART.
+            '''
+      features: [
+        "CSRNG.INTERFACE.HARDWARE0",
+        "CSRNG.INTERFACE.HARDWARE1",
+        "CSRNG.MODE.NONDETERMINISTIC",
+      ]
+      stage: V2
+      si_stage: SV2
+      tests: ["chip_sw_entropy_src_csrng"]
+    }
+    {
+      name: chip_sw_csrng_fuse_en_sw_app_read
+      desc: '''Verify the fuse input to CSRNG.
+
+            - Initialize the OTP with the fuse that control whether the SW can read the CSRNG state
+              enabled.
+            - Issue an instantiate command to request entropy.
+            - Verify that SW can read the internal state values.
+            - Reset the chip and repeat the steps above, but this time, initialized the OTP with
+              fuse disabled.
+            - Verify that the SW reads back all zeros when reading the internal state values.
+            '''
+      stage: V2
+      tests: ["chip_sw_csrng_fuse_en_sw_app_read_test"]
+    }
+    {
+      name: chip_sw_csrng_lc_hw_debug_en
+      desc: '''Verify the effect of LC HW debug enable on CSRNG.
```

Comment:
I know this test has been defined before already but I think 1) it would be worth to add a functional feature for this and 2) we should also run this for SiVal using the real silicon.

Reply:
Thanks for this feedback. I had initially not done this because it requires changes in the life-cycle stages. We can perform these tests at secure sites, so I have a updated the test plan to do this test at SV3.